manager2/src/app/tps/tps.component.ts

```diff
@@ -48,17 +53,27 @@ export class TpsComponent implements OnInit {
         private tpService: TpserviceService
     ) { }
 
+    private choseColor(id: string, over: boolean, created: boolean) {
+        let baseColor;
+        if (over) { baseColor = eventColors.over; }
+        else if (created) { baseColor = eventColors.created; }
+        else { baseColor = eventColors.pending; }
+        let color = { ...baseColor }
+        const hash = id.split('').reduce((acc, char) => acc + char.charCodeAt(0), 0);
+        const red = (hash % 128).toString(16).padStart(2, '0');
+        color.primary = `#${red}${color.primary.slice(3)}`;
+        return color;
+    }
```

Comment:
I fell like the color should be variable only for 'active' event. We can keep the same color for all the other events.
How variable are the colors?

Reply:
On askomics we use [this](https://github.com/askomics/flaskomics/blob/dev/askomics/react/src/classes/utils.jsx#L53) to generate the color, not sure if it's 'better'

Reply:
Here is the way it looked before:
![TpCalendar1](https://github.com/genouest/genouestaccountmanager/assets/95074810/02aa2687-2928-4d46-8fe5-447f3a5471f9)

Here it is with the color scheme you are recommending:
![TpCalendar2](https://github.com/genouest/genouestaccountmanager/assets/95074810/998ab555-926a-4d93-8700-970ecee5fefc)

This is way better indeed. Any other improvements ?